applications/tari_dan_app_utilities/src/template_manager/implementation/cmap_semaphore.rs

```diff
@@ -0,0 +1,67 @@
+//   Copyright 2023 The Tari Project
+//   SPDX-License-Identifier: BSD-3-Clause
+
+use std::{
+    fmt::{Debug, Formatter},
+    hash::Hash,
+    sync::{Arc, Mutex, MutexGuard},
+};
+
+#[derive(Clone)]
+pub struct CMapSemaphore<K: Hash + Eq> {
```

Comment:
```suggestion
pub struct ConcurrentMapSemaphore<K: Hash + Eq> {
```

Reply:
I would prefer a full name here to avoid confusion with C********* types that represent native objects. E.g. CString

Reply:
:100: